### DISCUSSION TITLE
Duplicity in combining UMAP with Mapper

### DISCUSSION THREAD

**Discussion by pranavkr29** [lmcinnes/umap#685] [Q&A] [answered]
I've been using Mapper(the scikit implentaion) with some of my datasets recently, and I was wondering if supplying UMAP as a lens to Mapper.map function(something which is quite common) is in some ways repeating some steps, since UMAP itself generates a Simplicial complex? I haven't had the time to dissect it fully, but could someone explain if that is the case ?

**Comment by lmcinnes** [comment, marked as answer]
In this sort of case Mapper is essentially providing a simplified version of the scatterplot UMAP might provide -- simplifying it via topology. Presuming you want that degree of simplification this is a perfectly acceptable thing to do. Note that Mapper often uses lenses such as PCA so using dimension reduction as a lens is pretty common. The real question is whether you need/want the degree of simplification that Mapper provides.

**Reply by pranavkr29** [reply]
Thanks for the answer. Could I bother you with a couple more? 

1) Is the UMAP function invertible? Meaning, if I pick the projected point, is it possible to apply an inverse transform to get back the original data point? Also, if I pick a point close to a projected point and use this inverse transform, will I get a point close to the original in the high dimensional point?
2) Unrelated to UMAP, but is there a way to estimate the joint PDF of a cluster in high dimension, something like KDE but based on the cluster rather than the individual points, such that we can then sample from the PDF ?

Thanks in advance.

**Reply by lmcinnes** [reply]
UMAP isn't invertible as a function, but you can solve an optimization problem to try to find an inverse point -- which is what the ``inverse_transform`` does. Alternatively if you use ParametricUMAP it learns an actual neural network transformation function and that can be inverted.

I'm not quite sure what you want or need with the PDF estimation. Density estimation in high dimensions is often difficult because of the curse of dimensionality. I'm not aware of any particularly strong methods.
